sky/clouds/service_catalog/kubernetes_catalog.py

```diff
@@ -26,3 +31,70 @@ def get_image_id_from_tag(tag: str, region: Optional[str]) -> Optional[str]:
 def is_image_tag_valid(tag: str, region: Optional[str]) -> bool:
     """Returns whether the image tag is valid."""
     return common.is_image_tag_valid_impl(_image_df, tag, region)
+
+
+def list_accelerators(
+        gpus_only: bool,
+        name_filter: Optional[str],
+        region_filter: Optional[str],
+        quantity_filter: Optional[int],
+        case_sensitive: bool = True
+) -> Dict[str, List[common.InstanceTypeInfo]]:
+    if Kubernetes not in global_user_state.get_enabled_clouds(
+    ) or not kubernetes_utils.check_credentials()[0]:
+        return {}
```

Comment:
Hmm, there seems to be a bug - my GKE cluster has V100 yet this shows the error message:

```
# sky show-gpus does not find GPUs:
(base) ➜  ~ sky show-gpus --cloud kubernetes
No GPUs found in Kubernetes cluster. If your cluster contains GPUs, make sure nvidia.com/gpu resource is available on the nodes and the node labels for identifying GPUs (e.g., skypilot.co/accelerators) are setup correctly. To further debug, run: sky check.


# Confirming sky launch detects GPUs:
(base) ➜  ~ sky launch -c test --gpus V100:1
I 10-05 14:31:23 optimizer.py:682] == Optimizer ==
I 10-05 14:31:23 optimizer.py:693] Target: minimizing cost
I 10-05 14:31:23 optimizer.py:705] Estimated cost: $0.0 / hour
I 10-05 14:31:23 optimizer.py:705]
I 10-05 14:31:23 optimizer.py:777] Considered resources (1 node):
I 10-05 14:31:23 optimizer.py:826] ------------------------------------------------------------------------------------------------------
I 10-05 14:31:23 optimizer.py:826]  CLOUD        INSTANCE           vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE     COST ($)   CHOSEN
I 10-05 14:31:23 optimizer.py:826] ------------------------------------------------------------------------------------------------------
I 10-05 14:31:23 optimizer.py:826]  Kubernetes   2CPU--8GB--1V100   2       8         V100:1         kubernetes      0.00          ✔
I 10-05 14:31:23 optimizer.py:826]  IBM          gx2-8x64x1v100     8       64        V100:1         us-east         2.50
I 10-05 14:31:23 optimizer.py:826]  GCP          n1-highmem-8       8       52        V100:1         us-central1-a   2.95
I 10-05 14:31:23 optimizer.py:826]  AWS          p3.2xlarge         8       61        V100:1         us-east-1       3.06
I 10-05 14:31:23 optimizer.py:826]  Azure        Standard_NC6s_v3   6       112       V100:1         eastus          3.06
I 10-05 14:31:23 optimizer.py:826] ------------------------------------------------------------------------------------------------------
I 10-05 14:31:23 optimizer.py:826]
Launching a new cluster 'test'. Proceed? [Y/n]: ^CAborted!


# Confirming the underlying code can detect GPUs:
(base) ➜  ~ SKYPILOT_DEBUG=1 python -c "import sky;print(sky.utils.kubernetes_utils.get_gpu_label_key_value('v100'))"
D 10-05 14:32:58 skypilot_config.py:157] Using config path: /Users/romilb/.sky/config.yaml
D 10-05 14:32:58 skypilot_config.py:160] Config loaded:
D 10-05 14:32:58 skypilot_config.py:160] None
D 10-05 14:32:58 skypilot_config.py:166] Config syntax check passed.
('cloud.google.com/gke-accelerator', 'nvidia-tesla-v100')
```

Reply:
This comparison is failing - `Kubernetes not in global_user_state.get_enabled_clouds()`. You may need to use `Kubernetes()` and the is_same_cloud comparator.

Reply:
Fixed in https://github.com/skypilot-org/skypilot/pull/2638/commits/2134873e47e43b5912c43d2fcf8b6812ac0d938b.
Checked both clusters with/without GPUs.